src/ray/raylet/agent_manager.cc

```diff
@@ -202,18 +178,25 @@ void AgentManager::CreateRuntimeEnv(
             callback(false, reply.serialized_runtime_env_context(),
                      /*setup_error_message*/ reply.error_message());
           }
-
         } else {
-          // TODO(sang): Invoke a callback if it fails more than X times.
-          RAY_LOG(INFO)
-              << "Failed to create the runtime env: " << serialized_runtime_env
-              << ", status = " << status
-              << ", maybe there are some network problems, will retry it later.";
+          std::stringstream stream;
+          stream << "Failed to create the runtime env: " << serialized_runtime_env
+                 << ", status = " << status
+                 << ", it is most likely the ray agent on this node is dead. Please "
+                    "check the log `dashboard_agent.log` on this node.";
+          if (retried_count >= kMaxRetry) {
```

Comment:
Without this, it is vulnerable to infinite hanging.

Reply:
I think this message is probably not necessary if agent fate shares with raylet (I don't think this can happen other than the situation agent is dead )